chapters/VK_EXT_present_timing/PresentTimeInfo.txt

```diff
@@ -0,0 +1,184 @@
+====
+
+include::{generated}/validity/structs/VkAbsolutePresentTimeEXT.txt[]
+
+--
+
+[open,refpage='VkRelativePresentTimeEXT',desc='Specifying the minimum duration an image should be presented',type='structs']
+--
+
+The sname:VkRelativePresentTimeEXT structure is defined as:
+
+include::{generated}/api/structs/VkRelativePresentTimeEXT.txt[]
+
+  * pname:presentID is an application-provided identification value.
+    If non-zero, flink:vkGetPastPresentationTimingEXT will return a
+    slink:VkPastPresentationTimingEXT struct with the timing information
+    associated with the present to the associated swapchain for this
+    call to flink:vkQueuePresentKHR, including this value being in the
+    pname:presentID member of the slink:VkPastPresentationTimingEXT
+    struct.
+    In order to be useful to the application, it should: be unique within
+    some period of time that is meaningful to the application.
+  * pname:minPresentDuration, if non-zero, specifies the minimum duration
+    in nanoseconds the application wants the image to be displayed to the
+    user.
+    A value of zero specifies that the presentation engine may: display the
+    image for any duration.
+  * pname:idealPresentDuration provides an indication to the presentation
+    engine of what the application desires its IPD to become.
+    An application can: set pname:idealPresentDuration to the same value
+    as pname:minPresentDuration, indicating that the application
+    desires its IPD to remain steady.
+    A value smaller than
+    pname:minPresentDuration indicates that the application desires to
+    shorten its IPD.
+    A value larger than pname:minPresentDuration
+    indicates that the application desires to lengthen its IPD.
+    The presentation engine will provide feedback on this value in the
+    data returned by flink:vkGetPastPresentationTimingEXT.
+  * pname:presentSlop is the period of time in nanoseconds before
+    pname:minPresentDuration elapses that the presentation engine may:
+    display the image.
+
+If pname:minPresentDuration is non-zero, the presentation engine should:
+display the image to the user for a minimum duration of
+(pname:minPresentDuration - pname:presentSlop) nanoseconds.
+
+Setting pname:minPresentDuration to zero is useful when the application
+desires to provide a non-zero pname:presentID in order to get back timing
+information from flink:vkGetPastPresentationTimingEXT, but does not wish
+to set a pname:minPresentDuration.
+
+Providing a pname:presentSlop enables the application to indicate to the
+presentation engine to display the image within this period if a vertical
+blanking period occurs on FRR displays.
+
+[NOTE]
+.Note
+====
+The pname:presentSlop is used to avoid unintentionally missing a vertical
+blanking period on FRR displays due to rounding errors or drift between
+clocks.
+A suggested value for pname:presentSlop is half the _Refresh Rate_.
```

Comment:
This rounding behavior feels a little inelegant. For most FRR scenarios, what is the motivation to use nanoseconds instead of a frame counter timebase?

Reply:
For the simple "glXSwapInterval" use case for example, the actual present times might not be that important, but it would be usetul to just spam a refresh cycle of 2 frames for example without having to compute and keep track of exact refresh durations, etc. Perhaps it would be possible to have a "frame count" timebase somehow?

Reply:
Relevant discussion in wayland-protocols: https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/45#note_627138

>having to compute and keep track of exact refresh durations

This doesn't sound like a lot of work tbh. Omitting the "frame count" timebase allows reducing the API surface and simplifying the extension.